drivers/soundwire/bus.c

```diff
@@ -842,15 +843,21 @@ static int sdw_slave_clk_stop_callback(struct sdw_slave *slave,
 				       enum sdw_clk_stop_mode mode,
 				       enum sdw_clk_stop_type type)
 {
-	int ret;
+	int ret = 0;
 
-	if (slave->ops && slave->ops->clk_stop) {
-		ret = slave->ops->clk_stop(slave, mode, type);
-		if (ret < 0)
-			return ret;
+	mutex_lock(&slave->sdw_dev_lock);
```

Comment:
is my understanding correct that every invocation of a slave drv->ops is going to require this mutex lock/unlock? Then, does it make sense to take the mutex inside the op instead of here?

Reply:
well, no. the mutex need to be taken before invoking the ops, because the driver can be removed and that would cause a deference of a dangling/NULL pointer.

That's the point of this whole series, making sure the ops pointer is always valid.